### PR TITLE
fix: use project level role to show warning on frontend

### PR DESCRIPTION
**Pull request by almeidabbm**
Closes: #10166 

### Description:

image

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by joaoviana** [comment]
Could you fix the type error on `build` @almeidabbm ? I will look at the PR soon!

File: packages/backend/src/models/SpaceModel.ts [hunk 1]
```diff
@@ -846,6 +851,7 @@ export class SpaceModel {
                             hasDirectAccess: !!user_with_direct_access,
                             inheritedRole: highestRole.role,
```
Comment (joaoviana) [inline packages/backend/src/models/SpaceModel.ts:852]: I believe this means that we don't use `inheritedRole` anymore. 
Could you update the `inheritedRole` to be `highestProjectRole?.role` or something similar? We don't need `projectRole` being added I think 🤔

**Review by joaoviana — COMMENTED** [review]
Everything else looks fine and tested! 
Just checking if we need an extra property, and then will approve 👍

File: packages/backend/src/models/SpaceModel.ts (hunk 1 above)
Comment (almeidabbm) [reply packages/backend/src/models/SpaceModel.ts:852]: I am seeing `inheritedRole` being used in `packages/frontend/src/components/common/ShareSpaceModal/ShareSpaceUserList.tsx` 😞

Comment (joaoviana) [reply packages/backend/src/models/SpaceModel.ts:852]: oops! ignore!

**Review by joaoviana — APPROVED** [review]

**Pull request merged**